cocos/renderer/gfx-gles2/GLES2Fence.cc

```diff
@@ -0,0 +1,49 @@
+#include "GLES2Std.h"
+#include "GLES2Fence.h"
+#include "GLES2Device.h"
+#include "GLES2GPUObjects.h"
+
+NS_CC_BEGIN
+
+GLES2Fence::GLES2Fence(GFXDevice* device)
+    : GFXFence(device)
+{
+}
+
+GLES2Fence::~GLES2Fence()
+{
+}
```

Comment:
It is better to invoke `destroy()`.

Reply:
I think we shouldn't do that here, the creator of GFX resources should be responsible for destroying it. Right now it's a matter of exit event (or lack of it there of), which should be passes to JS so that it can destroy them by the right time.

Reply:
Yep, indeed JS should make things work as expected. But invoke destroy() here doesn't make performance issue if it is already destroyed, and can make sure it is destroyed if it is not destroyed by JS to avoid memory leak.

I think the class or function itself make things more robust by itself as possible is a good practice.

Reply:
Generally yes, but then there's the destroy order issue, the gfx device & context should always be destroy last, relying on destructors to do this will immediately lead to many validation errors in vulkan backend.